src/test/scala/za/co/absa/hyperdrive/trigger/api/rest/services/WorkflowValidationServiceTest.scala

```diff
@@ -125,17 +125,86 @@ class WorkflowValidationServiceTest extends AsyncFlatSpec with Matchers with Moc
+    // given
+    val mapEmpty = Map.empty[String, List[String]]
+    val mapOne = Map("aa" -> List("11", "22"), "bb" -> List("33"))
+    val mapTwo = Map("bb" -> List("33"), "aa" -> List("11", "22"))
+    val mapThree = Map("cc" -> List("44", "55"))
+
+    val underTest = new WorkflowValidationServiceImpl(workflowRepository)
+
+    // then
+    underTest.areMapsEqual(mapEmpty, mapEmpty) shouldBe true
+    underTest.areMapsEqual(mapEmpty, mapOne) shouldBe false
+    underTest.areMapsEqual(mapOne, mapEmpty) shouldBe false
+    underTest.areMapsEqual(mapOne, mapThree) shouldBe false
+    underTest.areMapsEqual(mapThree, mapOne) shouldBe false
+    underTest.areMapsEqual(mapOne, mapOne) shouldBe true
+    underTest.areMapsEqual(mapOne, mapTwo) shouldBe true
+    underTest.areMapsEqual(mapTwo, mapOne) shouldBe true
+    underTest.areMapsEqual(mapThree, mapThree) shouldBe true
+  }
+
+  "areMapsOfMapsEqual" should "return true if maps contain same elements otherwise false" in {
+    // given
+    val mapEmpty = Map.empty[String, SortedMap[String, String]]
+    val mapOne = Map("aa" -> SortedMap("11" -> "22", "33" -> "44"), "bb" -> SortedMap("55" -> "66"))
+    val mapTwo = Map("bb" -> SortedMap("55" -> "66"), "aa" -> SortedMap("11" -> "22", "33" -> "44"))
+    val mapThree = Map("cc" -> SortedMap("77" -> "88"))
+
+    val underTest = new WorkflowValidationServiceImpl(workflowRepository)
+
+    // then
+    underTest.areMapsOfMapsEqual(mapEmpty, mapEmpty) shouldBe true
+    underTest.areMapsOfMapsEqual(mapEmpty, mapOne) shouldBe false
+    underTest.areMapsOfMapsEqual(mapOne, mapEmpty) shouldBe false
+    underTest.areMapsOfMapsEqual(mapOne, mapThree) shouldBe false
+    underTest.areMapsOfMapsEqual(mapThree, mapOne) shouldBe false
+    underTest.areMapsOfMapsEqual(mapOne, mapOne) shouldBe true
+    underTest.areMapsOfMapsEqual(mapOne, mapTwo) shouldBe true
+    underTest.areMapsOfMapsEqual(mapTwo, mapOne) shouldBe true
+    underTest.areMapsOfMapsEqual(mapThree, mapThree) shouldBe true
+  }
+
+  "validateWorkflowData" should "return empty seq when workflows are different otherwise seq with error" in {
+    // given
+    val originalWorkflow = WorkflowFixture.createWorkflowJoined();
+    val changeInDetails = originalWorkflow.copy(name = "differentName")
+    val changeInSensor = originalWorkflow.copy(
+      sensor = originalWorkflow.sensor.copy(
+        properties = originalWorkflow.sensor.properties.copy(matchProperties = Map("diffKey" -> "diffValue"))
+      )
+    )
+    val changeInJobs = originalWorkflow.copy(
+      dagDefinitionJoined = originalWorkflow.dagDefinitionJoined.copy(
+        jobDefinitions = Seq(
+          originalWorkflow.dagDefinitionJoined.jobDefinitions.head,
+          originalWorkflow.dagDefinitionJoined.jobDefinitions.last.copy(
+            name = "differentName"
+          )
+        )
+      )
+    )
+
+    val underTest = new WorkflowValidationServiceImpl(workflowRepository)
+
```

Comment:
okay, I would have created four different test cases but this is great to see 👍🏽.